enterprise/server/registry/registry_server.go

```diff
@@ -38,6 +40,13 @@ func (t *RegistryServer) Start() {
 	proxy := httputil.NewSingleHostReverseProxy(url)
 
 	http.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+		urlParts := strings.Split(req.URL.Path, "/")
+
+		if len(urlParts) > 3 && urlParts[1] == "modules" && strings.HasPrefix(urlParts[3], "github.") {
+			handleGitHub(w, req)
```

Comment:
up to you if it's worth it but you could have your top level method (or even this callsite) handle the reading and response writing and then your functional code that actually does a thing can return standard types and is more easily testable. totally a matter of preference tho

Reply:
Done